src/firstvoices/backend/tests/test_dictionary/test_parts_of_speech_view.py

```diff
@@ -0,0 +1,114 @@
+        response = client.get(reverse(self.API_LIST_VIEW, current_app=self.APP_NAME))
+        content = json.loads(response.content)["results"]
+
+        assert response.status_code == 200
+        assert len(content) > 0
+
+    def test_retrieve_base_case(self, db, parts_of_speech_db_setup):
+        """Test case for retrieve case to verify different querysets working for list and detail view."""
+
+        client = APIClient()
+        member_user = get_app_admin(AppRole.STAFF)
+        client.force_authenticate(user=member_user)
+        response_list = client.get(
+            reverse(
+                self.API_LIST_VIEW,
+                current_app=self.APP_NAME,
+            ),
+            format="json",
+        )
+        content_list = json.loads(response_list.content)["results"]
+
+        # Looking for a specific item with no children, to test the retrieve queryset
+        # test will fail if fixture not loaded since there will be no items without any children in the response to
+        # choose the first element from. If fixture is modified in such as way that all elements now have children at
+        # top level, Create another fixture for testing purposes with one such element.
+        items = [entry for entry in content_list if len(entry["children"]) == 0]
+        item = items[0]
+        pk = item["id"]
+
+        response = client.get(
+            reverse(self.API_DETAIL_VIEW, current_app="backend", args=[pk]),
+            format="json",
+        )
+        response_obj = json.loads(response.content)
+
+        assert response.status_code == 200
+        assert "id" in response_obj
+        assert "title" in response_obj
+        assert "children" in response_obj
+
+        assert isinstance(response_obj["id"], str)
+        assert isinstance(response_obj["title"], str)
+        assert isinstance(response_obj["children"], list)
+
+    def test_retrieve_not_found(self, db, parts_of_speech_db_setup):
+        # Testing the view returns 404 if parts of speech not found
+
+        client = APIClient()
+        member_user = get_app_admin(AppRole.STAFF)
+        client.force_authenticate(user=member_user)
+        response = client.get(
+            reverse(
+                self.API_DETAIL_VIEW,
+                current_app=self.APP_NAME,
+                args=["123_not_valid_key"],
+            ),
+            format="json",
+        )
+        assert response.status_code == 404
+
+    def test_response_format_list_view(self, db, parts_of_speech_db_setup):
+        # Testing the format of the content returned by list view for parts of speech
+
+        client = APIClient()
+        member_user = get_app_admin(AppRole.STAFF)
+        client.force_authenticate(user=member_user)
+        response = client.get(reverse(self.API_LIST_VIEW, current_app=self.APP_NAME))
+        content = json.loads(response.content)["results"]
+
+        assert isinstance(content, list)
+        assert len(content) > 0
+
+        print(content)
```

Comment:
This print statement should probably be removed.

Reply:
Yes, my bad. I suggest we should also add a check in the pre-commit hooks to not let this go through.